src/Services/OrderService/OrderService.Infrastructure/Consumers/PaymentStatusUpdatedConsumer.cs

```diff
@@ -0,0 +1,83 @@
+﻿using MassTransit;
+using Microsoft.Extensions.Logging;
+using OrderService.Application.Common.Extensions;
+using OrderService.Domain.Entities;
+using OrderService.IntegrationEvents;
+using PaymentService.IntegrationEvents;
+using OrderStatus = OrderService.IntegrationEvents.OrderStatus;
+
+namespace OrderService.Infrastructure.Consumers;
+
+public class PaymentStatusUpdatedConsumer : IConsumer<PaymentRequestProcessedEvent>
+{
+    private readonly ILogger<PaymentStatusUpdatedConsumer> _logger;
+    private readonly IPublishEndpoint _publishEndpoint;
+    private readonly ApplicationDbContext _context;
+
+    public PaymentStatusUpdatedConsumer(ILogger<PaymentStatusUpdatedConsumer> logger,
+        IPublishEndpoint publishEndpoint,
+        ApplicationDbContext dbContext)
+    {
+        _logger = logger;
+        _publishEndpoint = publishEndpoint;
+        _context = dbContext;
+    }
+
+    public async Task Consume(ConsumeContext<PaymentRequestProcessedEvent> context)
+    {
+        var message = context.Message;
+
+        var paymentStatus = PaymentStatusMapperExtensions.ToDomainPaymentStatus(message.PaymentStatus);
```

Comment:
this is extension method, use it as extension: message.PaymentStatus.ToDomainPaymentStatus()